In [ ]:
install.packages("tidyverse")
install.packages("skimr")
library(tidyverse)
library(skimr)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Warning message in system("timedatectl", intern = TRUE):
“running command 'timedatectl' had status 1”
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6      ✔ purrr   0.3.4 
✔ tibble  3.1.7      ✔ dplyr   1.0.10
✔ tidyr   1.2.1      ✔ stringr 1.4.1 
✔ readr   2.1.2      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


In [ ]:
hotels <- read.csv("hotels.csv")

In [ ]:
#1
skim(hotels) #shows all the information

1. skim(hotels) gives me the data summary and the column type frequency. The data summary contains the name of the data, number of columns and rows in the data, and the column type frequency contains character and numeric. It helps me understand the information of people who stay in each hotel, and it tells whether the data has missing data or not.


In [ ]:
#2
hotels %>%
  filter(
    country != "USA",
    lead_time < 1
  ) %>% #removing all the visitors from the US
  select(lead_time) #shows the lead_time


2. The command filters all the visitors from the USA, therefore, the data only provides the visitors from the outside of the country. Also, we got rid of all the data that the lead time are less than a day. The data frame’s number decreased from 119390 to 6174.


In [ ]:
#3
hotels %>%
  filter(
    children >= 1| babies >= 1
  ) #filtering all the visitors with at least one child or baby



3. The command gets rid of all the visitors  with no children or babies . The data frame’s number decreased from 119390 to 9332.


4. I think it is more likely to find bookings with children or babies in resort hotels.


In [ ]:
#5
hotels %>%
  filter(
    children >= 1| babies >= 1,
    hotel == "Resort Hotel"
  ) %>%#filtering all the visitors with 0 children or babies
  select(hotel,adults, babies, children)


hotel        adults babies children adr   
1    Resort Hotel 2      0      1        154.77
2    Resort Hotel 2      0      2        153.00
3    Resort Hotel 2      0      2        107.00
4    Resort Hotel 2      0      2        153.00
5    Resort Hotel 2      0      1        108.73
6    Resort Hotel 2      0      1         99.30
7    Resort Hotel 1      0      2         96.49
8    Resort Hotel 2      0      2        163.00
9    Resort Hotel 2      1      0         73.80
10   Resort Hotel 2      1      0        139.00
11   Resort Hotel 2      0      1        149.00
12   Resort Hotel 2      0      2        163.00
13   Resort Hotel 2      0      1         85.59
14   Resort Hotel 2      0      2        163.00
15   Resort Hotel 2      0      2        164.20
16   Resort Hotel 2      0      2        175.00
17   Resort Hotel 2      0      2        167.00
18   Resort Hotel 2      0      2        210.00
19   Resort Hotel 2      1      0         97.00
20   Resort Hotel 2      0      1        131.00
21   Resort Hotel 2      0      2        185.00
22   Resort Hotel 2      0      1        185.50
23   Resort Hotel 3      0      2        193.00
24   Resort Hotel 2      0      1        112.00
25   Resort Hotel 2      0      2        123.00
26   Resort Hotel 2      0      2        146.00
27   Resort Hotel 2      0      2        123.00
28   Resort Hotel 2      0      2        153.00
29   Resort Hotel 2      1      0         82.00
30   Resort Hotel 2      0      1        185.00
⋮    ⋮            ⋮      ⋮      ⋮        ⋮     
3900 Resort Hotel 2      0      2        210.00
3901 Resort Hotel 2      0      1        186.00
3902 Resort Hotel 2      0      1        219.03
3903 Resort Hotel 2      0      2        308.57
3904 Resort Hotel 2      0      1        277.71
3905 Resort Hotel 2      0      2        336.57
3906 Resort Hotel 2      0      2        289.86
3907 Resort Hotel 2      1      0        124.54
3908 Resort Hotel 2      1      0        123.00
3909 Resort Hotel 2      0      1        153.75
3910 Resort Hotel 2      0      2        298.00
3911 Resort Hotel 2      0      2        225.20
3912 Resort Hotel 1      0      2        284.29
3913 Resort Hotel 3      0      1        383.00
3914 Resort Hotel 2      0      2        200.00
3915 Resort Hotel 2      0      1        176.25
3916 Resort Hotel 2      0      1        250.20
3917 Resort Hotel 2      1      0        169.60
3918 Resort Hotel 2      0      2        232.50
3919 Resort Hotel 2      0      1        230.43
3920 Resort Hotel 2      0      1        230.43
3921 Resort Hotel 2      0      1        207.50
3922 Resort Hotel 2      0      1        178.58
3923 Resort Hotel 2      0      1        147.01
3924 Resort Hotel 2      0      2        222.86
3925 Resort Hotel 2      0      1         87.75
3926 Resort Hotel 2      0      1        312.29
3927 Resort Hotel 2      0      1        127.58
3928 Resort Hotel 2      0      1        125.58
3929 Resort Hotel 2      0      1         89.75

5. The number of bookings in resort hotels that have more than one child or one baby in the room is 3929.

In [ ]:
#6
hotels %>%
  filter(
    children >= 1| babies >= 1,
    hotel == "City Hotel"
  ) %>%#filtering all the visitors with 0 children or babies
  select(hotel,adults, babies, children)


6. The number of bookings in city hotels that have more than one child or one baby in the room is 5403.

In [ ]:
#7
hotels %>%
  filter(
    children >= 1| babies >= 1,
  ) %>%#filtering all the visitors with 0 children or babies
  select(hotel,adults, babies, children)

7. We find more in the city hotel than the resort hotel.


In [ ]:
#8
hotels %>%
  count(adults, sort = TRUE)

8. The party of 2 are the most common number of adults in bookings. 89680 parties. It is surpring that there are 403 bookings with 0 adults.

In [ ]:
#9
hotels %>%
  count(is_canceled, sort = TRUE)

is_canceled n    
1 0           75166
2 1           44224

9. Around the quater of the bookings were canceled.

In [ ]:
#10
hotels %>%
  filter(
    hotel == "Resort Hotel"
  ) %>%
  summarise(
    min_adr = min(adr),
    mean_adr = mean(adr),
    median_adr = median(adr),
    max_adr = max(adr)
  )



min_adr mean_adr median_adr max_adr
1 -6.38   94.95293 75         508

In [ ]:
#10
hotels %>%
  filter(
    hotel == "City Hotel"
  ) %>%
  summarise(
    min_adr = min(adr),
    mean_adr = mean(adr),
    median_adr = median(adr),
    max_adr = max(adr)
    )


min_adr mean_adr median_adr max_adr
1 0       105.3045 99.9       5400

10. City hotel is higher on average.

11. The resort hotels has a negative minimum (-6.38), and the city hotel has a very high maximum (5400).

In [ ]:
hotels %>%
  filter(
    adr == -6.38 | adr == 5400
  ) %>%
  select(adr, arrival_date_year, arrival_date_month, adults, children, babies)

adr     arrival_date_year arrival_date_month adults children babies
1   -6.38 2017              March              2      0        0     
2 5400.00 2016              March              2      0        0